enhanced-resolve/lib/Storage.d.ts

```diff
@@ -0,0 +1,34 @@
+/// <reference types="node" />
+import { CommonFileSystemMethod } from './common-types';
+
+declare class Storage {
+    duration: number;
+    running: {
+        [name: string]: ((this: null, ...args: any[]) => any)[]
+    };
+    data: {
+        [name: string]: any[]
+    };
+    levels: string[][];
+    count: number;
+    interval: NodeJS.Timer | null;
+    needTickCheck: boolean;
+    nextTick: number | null;
+    passive: boolean;
+
+    constructor(duration: number);
+
+    ensureTick(): void;
+
+    finished(name: string): void;
+
+    provide(name: string, provider: CommonFileSystemMethod, callback: (this: null, ...args: any[]) => any): any;
```

Comment:
Why does this need `this: null`? Generally if you're not providing a `this` value you can just not specify it. E.g. in `Array` we have `sort(compareFn?: (a: T, b: T) => number): this;`, not `(this: null, a: T, b: T) => number`.